src/main/java/com/es/phoneshop/dao/CustomProductDao.java

```diff
@@ -70,6 +53,21 @@ public Optional<Product> getProduct(Long id) {
         return Optional.empty();
     }
 
+    @Override
+    public Optional<Product> getProductByCode(String code) {
```

Comment:
our code is not unique filed, that means that we can have 2 products with the same code
pdp result may be unexpected to the customer